### ISSUE TITLE
rrdcached plugin: rrdc_update failed with status -1

### ISSUE THREAD

**Issue by rubenk** [collectd/collectd#1947]
I just upgraded my collectd clients and servers to 5.6.0
I had to revert because my error logs were flooded.

I did see this on earlier versions sporadically, and I worked around it by setting WriteThreads to 1, but now it happens much more often.

Let me know what debug info I can provide.

*   Version of collectd:

5.6.0

*   Operating system / distribution:

CentOS 6 and CentOS 7

## Expected behavior

No errors

(Description of the behavior / output that you expected)

## Actual behavior

Error logs fill up fast with
```
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: Filter subsystem: Built-in target `write': Dispatching value to all write plugins failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_number-isolated_file; value time = 1474292082.898; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_number-isolated_file.rrd, [1474292083:0.000000], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_number-dirty; value time = 1474292082.898; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_number-dirty.rrd, [1474292083:0.000000], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_number-shmem; value time = 1474292082.898; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_number-shmem.rrd, [1474292083:42.000000], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_number-anon_transparent_hugepages; value time = 1474292082.898; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_number-anon_transparent_hugepages.rrd, [1474292083:0.000000], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_number-slab_unreclaimable; value time = 1474292082.898; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_number-slab_unreclaimable.rrd, [1474292083:14430.000000], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_number-inactive_file; value time = 1474292082.898; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_number-inactive_file.rrd, [1474292083:14351.000000], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_faults; value time = 1474292082.906; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_faults.rrd, [1474292083:2441847244:898], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_io-memory; value time = 1474292082.906; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_io-memory.rrd, [1474292083:251939:8056416], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: uc_update: Value too old: name = queue1b.tilaa.nl/vmem/vmpage_io-swap; value time = 1474292082.906; last cache update = 1474292125.426;
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: rrdcached plugin: rrdc_update (/collectd/rrd/queue1b.tilaa.nl/vmem/vmpage_io-swap.rrd, [1474292083:8:400], 1) failed with status -1.
Sep 19 15:35:42 evcollectd1a.tilaa.nl collectd[28432]: Filter subsystem: Built-in target `write': Some write plugin is back to normal operation. `write' succeeded.

```

This happens for various plugins.

**Comment by octo** [comment]
Hi @rubenk,

are these metrics received by the *network plugin*, send to this instance by another mean or gathered locally?

Best regards,
—octo

**Comment by rubenk** [comment]
Hi @octo,

Sorry, I should have mentioned that. Yes, I have one server receiving metrics via the network plugin and updating rrd's with the rrdcached plugin. These logs are from that server.

**Comment by rubenk** [comment]
This looks like a PEBKAC, sorry. Since reverting to 5.5.2 didn't help I dug deeper. I use my own build of the collectd 5 rpm for CentOS 6. While updating that to 5.6.0 I switched the init script it used from contrib/fedora/ to contrib/redhat since we removed the former. The latter init script uses collectdmon, which started collectd. This resulted in collectd running twice on my clients, hence the issue.
